runtime/src/bank.rs

```diff
@@ -12525,101 +12525,133 @@ pub(crate) mod tests {
+        bank2.transfer(9, &key5, &key3.pubkey()).unwrap();
+
         bank2.freeze(); // the freeze here is not strictly necessary, but more for illustration
         bank2.squash();
 
         drop(bank1);
-
         bank2.clean_accounts(false);
+
+        let expected_ref_count_for_cleaned_up_keys = 0;
+        let expected_ref_count_for_keys_only_in_slot_2 = bank2
+            .rc
+            .accounts
+            .accounts_db
+            .accounts_index
+            .ref_count_from_storage(&key2.pubkey());
+
         assert_eq!(
             bank2
                 .rc
                 .accounts
                 .accounts_db
                 .accounts_index
-                .ref_count_from_storage(&pubkey1),
-            0
+                .ref_count_from_storage(&key1.pubkey()),
+            expected_ref_count_for_cleaned_up_keys
         );
-    }
-
-    #[test]
-    fn test_clean_dropped_unrooted_unfrozen_banks() {
-        //! Test that dropped unrooted, unfrozen banks are cleaned up properly
-        //!
-        //! slot 0:       bank0 (rooted)
-        //!               /   \
-        //! slot 1:      /   bank1 (unrooted, unfrozen, and dropped)
-        //!             /
-        //! slot 2:  bank2 (rooted)
-        //!
-        //! In the scenario above, when `clean_accounts()` is called on bank2, the keys that exist
-        //! _only_ in bank1 should be cleaned up, since those keys are unreachable.
-        //
-        solana_logger::setup();
-
-        let (genesis_config, mint_keypair) = create_genesis_config(100);
-        let bank0 = Arc::new(Bank::new(&genesis_config));
-
-        let collector = Pubkey::new_unique();
-        let pubkey1 = Pubkey::new_unique();
-        let pubkey2 = Pubkey::new_unique();
-
-        bank0.transfer(2, &mint_keypair, &pubkey2).unwrap();
-        bank0.freeze();
-
-        let slot = 1;
-        let bank1 = Bank::new_from_parent(&bank0, &collector, slot);
-        bank1.transfer(3, &mint_keypair, &pubkey1).unwrap();
-        // bank1 is not frozen on purpose
-
-        let slot = slot + 1;
-        let bank2 = Bank::new_from_parent(&bank0, &collector, slot);
-        bank2.transfer(4, &mint_keypair, &pubkey2).unwrap();
-        bank2.freeze(); // the freeze here is not strictly necessary, but more for illustration
-        bank2.squash();
-
-        drop(bank1);
-
-        bank2.clean_accounts(false);
         assert_eq!(
             bank2
                 .rc
                 .accounts
                 .accounts_db
                 .accounts_index
-                .ref_count_from_storage(&pubkey1),
-            0
+                .ref_count_from_storage(&key3.pubkey()),
+            expected_ref_count_for_keys_only_in_slot_2
+        );
+        assert_eq!(
+            bank2
+                .rc
+                .accounts
+                .accounts_db
+                .accounts_index
+                .ref_count_from_storage(&key4.pubkey()),
```

Comment:
Checking the ref count to verify zero-lamport accounts is not a bad idea, but I think an additional top-level should also be made to make sure it was actually removed from the index. 

I recommend something like this check used in other zero lamport clean tests: https://github.com/solana-labs/solana/blob/master/runtime/src/accounts_db.rs#L6626-L6629

Reply:
Done!